ADMIN-PORT.md

```diff
@@ -0,0 +1,45 @@
+---
+RFC: 3
+Status: Proposed
+---
+
+# ADMIN PORT RFC
+
+## Abstract
+
+In Valkey.conf, Add a new config parameter: "admin-port" to allow internal administrators in a company or special users from public internet to connect to the Valkey server to execute some special commands or load specfic modules.
+
+
+## Motivation
+
+For adminstrators, they may want to execute some special commands in the server, and other general clients are not allowed to run these kinds of commands.
+We want to let adminstrator connect to this admin port to do some special things. 
```

Comment:
There is something here that I still don't understand.

Which *special commands* exactly are you thinking about? An administrator can start the node with a config file or write the config on the command line. The administrator has access to the machine with ssh or something like that, right?

Is it something else you want to do?

Do you want to start all nodes in the same way, and later configure them for each customer?

Do you want the customer to have full access to all commands and even configure ACL, but still not be able to access something that only you (cloud operator) can access using the admin port?

Reply:
![image](https://github.com/user-attachments/assets/75d75731-1888-4759-9d04-fe573c4f10c9)

Reply:
>However sometimes, admin do not want to this (maybe due to the uncertain username),

Could you elaborate on why they would like to avoid using ACLs? 

My guess is that rotating secrets adds complexity and management overhead. Without regular rotation, the password or secret becomes vulnerable to compromise. In that case, security is weakened because a static secret is bound to leak over time.

If that's the case, I could be convinced to implement a dedicated port restricted to the loopback interface. However, I think it should still integrate with the ACL subsystem. Specifically, we may need to introduce a hardcoded `admin` user with default ACL settings that can be overridden via configuration. Is that in line with your thinking?

Reply:
Perhaps use a dedicated Unix socket instead of an actual port and leave it to the customer to decide how to access it (it can be forwarded to a TCP port).